prow/test/integration/test/crier_test.go

```diff
@@ -0,0 +1,187 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowgh "k8s.io/test-infra/prow/github"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestReportGHStatus(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		states []prowjobv1.ProwJobState
+		want   []string
+	}{
+		{
+			name: "triggered",
+			want: []string{"pending"},
+		},
+		{
+			name:   "pending",
+			states: []prowjobv1.ProwJobState{prowjobv1.PendingState},
+			want:   []string{"pending", "pending"},
+		},
+		{
+			name:   "pending-success",
+			states: []prowjobv1.ProwJobState{prowjobv1.SuccessState},
+			want:   []string{"pending", "success"},
+		},
+		{
+			name:   "pending-failed",
+			states: []prowjobv1.ProwJobState{prowjobv1.FailureState},
+			want:   []string{"pending", "failure"},
+		},
+		{
+			name:   "pending-aborted",
+			states: []prowjobv1.ProwJobState{prowjobv1.AbortedState},
+			want:   []string{"pending", "failure"},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			clusterContext := getClusterContext()
+			t.Logf("Creating client for cluster: %s", clusterContext)
+
+			kubeClient, err := NewClients("", clusterContext)
+			if err != nil {
+				t.Fatalf("Failed creating clients for cluster %q: %v", clusterContext, err)
+			}
+
+			ctx := context.Background()
+
+			podName := fmt.Sprintf("crier-test-pod-%s", RandomString(t))
+			sha := RandomString(t)
+
+			t.Logf("Creating CRD %s for sha %s", podName, sha)
+			prowjob := prowjobv1.ProwJob{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{
+						"prow.k8s.io/job":       podName,
+						"prow.k8s.io/refs.org":  "fake-org",
+						"prow.k8s.io/refs.pull": "0",
+						"prow.k8s.io/refs.repo": "fake-repo",
+						"prow.k8s.io/type":      "presubmit",
+					},
+					Labels: map[string]string{
+						"created-by-prow":  "true",
+						"prow.k8s.io/type": "presubmit",
+						"name":             podName,
+					},
+					Name:      podName,
+					Namespace: defaultNamespace,
+				},
+				Spec: prowjobv1.ProwJobSpec{
+					Type:      prowjobv1.PresubmitJob,
+					Namespace: "test-pods",
+					Job:       podName,
+					Refs: &prowjobv1.Refs{
+						Org:     "fake-org",
+						Repo:    "fake-repo",
+						BaseRef: "master",
+						BaseSHA: "49e0442008f963eb77963213b85fb53c345e0632",
+						Pulls: []prowjobv1.Pull{
+							{
+								Author: "fake_author",
+								Number: 0,
+								SHA:    sha,
+							},
+						},
+					},
+					Report: true,
+				},
+				Status: prowjobv1.ProwJobStatus{
+					State: prowjobv1.TriggeredState,
```

Comment:
Instead of implicitly testing the `Triggered` state in every job, why don't we do that explicitly with a testcase?

Reply:
Updated to do just one state per testcase